### PR TITLE
doc: Software version proposal

### PR DESCRIPTION
**Pull request by tbruyelle**
[Rendered version](https://github.com/tbruyelle/govgen-proposals/blob/tbruyelle/doc/version-proposal/002_SOFTWARE_VERSIONING.md)

### CODE REVIEW

**Review by jaekwon — APPROVED** [review]

File: 002_SOFTWARE_VERSIONING.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+### Cosmos-SDK
+
+The more important part of the dependency is the Cosmos-SDK, so it is the first
+version we need to define.
+
+While v0.45 is a battle-tested version and has proven to be viable, an [audit
+driven by OAK Security][audit] in early 2024 has reported some serious threats
+that have been fixed in v0.47. 
+
+On the other hand, v0.50 brings some interesting features in term of
```
Comment (giunatale) [inline 002_SOFTWARE_VERSIONING.md:50]:
```suggestion
On the other hand, while v0.50 brings some interesting features in term of
```

File: 002_SOFTWARE_VERSIONING.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+modularity, but it's not used by the CosmosHub yet (at the time of writing,
```
Comment (giunatale) [inline 002_SOFTWARE_VERSIONING.md:51]:
```suggestion
modularity, it is not used by the CosmosHub yet (at the time of writing,
```

File: 002_SOFTWARE_VERSIONING.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+Gaia v19 upgrade proposal is still in voting period).
```
Comment (giunatale) [inline 002_SOFTWARE_VERSIONING.md:52]: Is this the only argument we want to bring for discarding 0.50.x? Might be challenged if put it this way. I think we need to put emphasis on the fact that newer versions have less guarantees in terms of reliability. Unless this was already discussed or decided to not mention, imho it should be.

File: 002_SOFTWARE_VERSIONING.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+With the defined version of the Cosmos-SDK, this limits the choice of Gaia
+codebase to fork to 4 versions:
+
+- v15 fixes all the issues reported by the OAK audit
+- v16 adds some IBC/ICS functionalities
+- v17 upgrades ICS to 2.0
+- v18 adds the wasmd and feemarket module
+
+Obviously, v18 is discarded because AtomOne won't have any VM functionality, as
```
Comment (giunatale) [inline 002_SOFTWARE_VERSIONING.md:72]:
```suggestion
AtomOne will not have any VM functionality as
```

File: 002_SOFTWARE_VERSIONING.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+specified in the draft of the [Constitution].
```
Comment (giunatale) [inline 002_SOFTWARE_VERSIONING.md:73]:
```suggestion
specified in the draft of the [Constitution], thus v18 gets automatically discarded.
```

File: 002_SOFTWARE_VERSIONING.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+Then, considering that the first version of AtomOne won't have IBC or ICS
```
Comment (giunatale) [inline 002_SOFTWARE_VERSIONING.md:75]:
```suggestion
Considering that the first version of AtomOne will not have IBC or ICS
```

File: 002_SOFTWARE_VERSIONING.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+enabled, there's no advantage in choosing to fork v16 or v17.
```
Comment (giunatale) [inline 002_SOFTWARE_VERSIONING.md:76]:
```suggestion
enabled, there is no advantage in choosing to fork v16 or v17.
```

File: 002_SOFTWARE_VERSIONING.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+This left out v15, which is the version we propose to fork for AtomOne.
```
Comment (giunatale) [inline 002_SOFTWARE_VERSIONING.md:78]:
```suggestion
This leaves out only v15, which is the version we propose to fork for AtomOne.
```

**Review by giunatale — CHANGES_REQUESTED** [review]

File: 002_SOFTWARE_VERSIONING.md (hunk 3 above)
Comment (tbruyelle) [reply 002_SOFTWARE_VERSIONING.md:52]: This is a good point, let's improve that part.

Comment (tbruyelle) [reply 002_SOFTWARE_VERSIONING.md]: @giunatale see aaa72667fb714518227a21776461fb6044e48342

**Review by giunatale — APPROVED** [review]

**Pull request merged**